src/Package/Fetch.zig

```diff
@@ -65,6 +65,9 @@ oom_flag: bool,
 /// If the resource pointed to by the location is not a Git-repository, this
 /// will be left unchanged.
 latest_commit: ?git.Oid,
+/// If `location` was `path_or_url`, this will be set to indicate
+/// whether the CLI input was identified as a file path or a URL.
+actual_path_or_url_kind: enum { path, url },
```

Comment:
I just want to note that this does mean that if the user invokes a command like `zig fetch --save ./foo.tar.gz`, the package will first be copied to the global cache before printing the error. This is obviously not perfect; ideally, the command should detect that the user passed a file path in combination with `--save` and exit before doing any work. However, if we want a maximally consistent and correct behavior, each solution comes with its own downsides

- Currently, we test if the `path_or_url` input is a directory we can open, then a file we can open, then finally a URL. Moving this logic to the `cmdFetch()` function before we actually fetch the package would mean we now have open file handles that we need to pass to the `run()` function and correctly close them if any errors occur on the way there (but not if `run()` returns an error since `run()` claims ownership of the handle). This is not trivial and will result in ugly code.
- We could change the order of operations and instead try parsing the input as an URL first. However, an input like `c:foo` (on Windows) or `http:foo` (on POSIX) would be interpreted as a URL despite possibly being a valid file path, so this isn't ideal. Even if we try to rule out values we know are not valid URLs that can be fetched, an input like `http://example.com/example.tar.gz` *could* be a valid file path to the subpath `example.com/example.tar.gz` inside the directory `http:`.
- We could also change the fetch command so that the user explicitly has to specify whether the input is a path or url, as in `zig fetch --path http:foo --save`.

I've thought about this a lot and I'm sure what the best way to handle this is, or if it's even worth putting energy into. I just think edge cases matter, however obscure they might be.